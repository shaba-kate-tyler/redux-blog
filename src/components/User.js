import React from 'react';
import PropTypes from 'prop-types';

function User({ name, username, website, email }) {
  return (
    <div>
      <h3>{name}</h3>
      <h4>{username}</h4>
      <p>{website} / {email}</p>
    </div>
  );
}

User.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  website: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired
};

export default User;
