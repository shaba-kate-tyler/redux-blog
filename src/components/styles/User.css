.User {
    text-align: center;
    text-decoration: none;
    font-family: 'Roboto', sans-serif;
    h3 {
        color: rgb(66,139,202);
        font-weight: 500;
        font-size: 4vh;
    }
    h4 {
        color: black;
        font-weight: 100;
    }
    p {
        color: black;
        font-weight: 100;
    }
}