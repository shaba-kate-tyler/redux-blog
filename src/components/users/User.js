import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../styles/User.css';

function User({ id, name, username, website, email }) {
  return (
    <Link to={`/users/${id}`} className={styles.User}>
      <div>
        <h3>{name}</h3>
        <h4>{username}</h4>
        <p>{website} / {email}</p>
      </div>
    </Link>
  );
}

User.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  website: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired
};

export default User;
