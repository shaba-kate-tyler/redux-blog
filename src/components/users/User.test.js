import React from 'react';
import User from './User';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

describe('User', () => {
  it('matches a snapshot', () => {
    const props = {
      id: 3,
      name: 'kaiya',
      username: 'ladybeard',
      website: 'schnepherds4life.org',
      email: 'itsme@kai.com'
    };

    const tree = renderer.create(
      <MemoryRouter>
        <User { ...props } />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
