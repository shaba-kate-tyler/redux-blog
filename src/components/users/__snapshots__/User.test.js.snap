// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a snapshot 1`] = `
<a
  className="User"
  href="/users/3"
  onClick={[Function]}
>
  <div>
    <h3>
      kaiya
    </h3>
    <h4>
      ladybeard
    </h4>
    <p>
      schnepherds4life.org
       / 
      itsme@kai.com
    </p>
  </div>
</a>
`;
